### ISSUE TITLE
usbipd latency

### ISSUE THREAD

**Issue by Jlo6CTEP** [dorssel/usbipd-win#636]
Hi everyone! 

Im currently working on a HITL simulator with PX4 / Cyphal stack. 
PX4 is an autopilot, and HITL basically means that you feed the autopilot sensory data generated with simulator (which is on PC) so that it thinks that it is flight and acts accordingly. 

Communication is done through custom USB-CAN adapter + SLCAN on top of that. WSL by default does not support SLCAN, so I had to build the kernel to enable slcan drivers. It worked, however, when I boot up my sim, it shows weird behavior similar to over-damping, rendering it impossible to control the drone.

I started diagnosing the problem with PX4 log analyzer and I believe that the issue boils down to high latency of usbipd bridge - on average, latency is anywhere between 20 and 120ms, while on native Linux it is around 15 ms
Native
![image](https://github.com/dorssel/usbipd-win/assets/35602222/20e3a496-de4c-44f8-becd-ef5fbc726397)
USBIPD
![image](https://github.com/dorssel/usbipd-win/assets/35602222/ce2417d0-8215-431d-b64f-428fff8363a5)
It looks like it works OK when you do things like flashing MCUs but struggles with many small packets in case of SLCAN

Is there a way to make the bridge  faster?
And by the way, thank you for this piece of software, it helped me countless times with other devices like depth cameras, flashing hardware etc, really cool stuff.

**Comment by dorssel** [comment]
> Is there a way to make the bridge faster?

Yes, but only slightly. See https://github.com/dorssel/usbipd-win/issues/469#issuecomment-1304316843.

Root cause is that `usbipd-win` is user mode software, and therefore requires several context switches between kernel and user mode for each USB request. That's not going to change any time soon; it would require putting all functionality, including network calls, into a driver to stay in kernel mode (this is how it works in Linux, BTW).
